src/lib/y2storage/md_container.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/storage_class_wrapper"
+require "y2storage/md"
+
+module Y2Storage
+  # MD Container
+  #
+  # This is a wrapper for Storage::MdContainer
```

Comment:
I would appreciate more documentation here about what an MdContainer is. The inheritance-based solution to distinguish software MD RAIDs from BIOS ones is counterintuitive to say the least. So in general I would add a couple of sentences to Md, another couple to MdContainer, same for MdMember and maybe a deeper explanation of `#software_defined?` and its implications.